tests/load/lancedb/test_unit_test_add_vector_column.py

```diff
@@ -0,0 +1,301 @@
+import pyarrow as pa
+from datetime import datetime, date
+from decimal import Decimal
+
+from dlt.destinations.impl.lancedb.schema import add_vector_column
+
+
+def test_add_vector_column_basic_types() -> None:
```

Comment:
I'm not sure what we are testing here. We have 4 cases to test:
- both schemas have vector column
- none has
- two other possibilities

the test should make sure that output schema has a vector column (added or kept) exactly at the index it was there originally. checking nulls is a nice addon :)

that should have end to end test as well where user adds vector in the input data. I wrote one: test_bring_your_own_vector I think it is enough for now. It would be good to have a test where we add vector ourselves but also ask one to be added with lance_adapter. but IMO we do not have time to do that now